components/sup/src/manager/service/mod.rs

```diff
@@ -18,7 +18,7 @@ mod dir;
 mod health;
 pub mod hooks;
 mod package;
-mod spec;
+pub mod spec;
```

Comment:
I'm trying to understand why adding the `pub` is necessary, and I think it's because the definition of `DesiredState` that you're referencing here is the one from [sup/src/manager/service/spec.rs](https://github.com/jeremymv2/habitat/blob/master/components/sup/src/manager/service/spec.rs#L46-L49) rather than the one from [sup-protocol/src/generated/sup.types.rs](https://github.com/jeremymv2/habitat/blob/master/components/sup-protocol/src/generated/sup.types.rs) as with [`ServiceGroup`](https://github.com/jeremymv2/habitat/blob/master/components/sup-protocol/src/generated/sup.types.rs#L61-L70). Is that intentional?

Reply:
@baumanj yes, this is result of this implementation, introducing another discrete `DesiredState` type. Now there's two, one from [protocol::types::DesiredState](https://github.com/jeremymv2/habitat/blob/jeremymv2/desired_state_for_svc_status/components/sup-protocol/src/generated/sup.types.rs#L113-L116) and the other from [manager::service::spec::DesiredState](https://github.com/jeremymv2/habitat/blob/jeremymv2/desired_state_for_svc_status/components/sup/src/manager/service/spec.rs#L46-L49)

As far as I can tell, `protocol::types::DesiredState` is necessary in order to get it into the [ServiceStatus](https://github.com/jeremymv2/habitat/blob/jeremymv2/desired_state_for_svc_status/components/sup-protocol/src/generated/sup.types.rs#L90-L91) proto message. 

For full disclosure I also have an implementation that removes `manager::service::spec::DesiredState` and just `protocol::types::DesiredState` remains. However, that required a lot of re-wiring and isn't completely working - with an unsolved bug where `hab svc stop` was not working (the spec watcher never detected a change). Thus, I went with the current implementation strategy.

With the current implementation there needs to be a conversion from `protocol::types::DesiredState` <-> `manager::service::spec::DesiredState` because ultimately, the source of truth is derived from the [spec](https://github.com/jeremymv2/habitat/blob/jeremymv2/desired_state_for_svc_status/components/sup/src/manager/service/mod.rs#L183)

To implement the [conversion](https://github.com/jeremymv2/habitat/blob/jeremymv2/desired_state_for_svc_status/components/sup/src/manager/mod.rs#L1944-L1951) 
I needed to make the `spec` module public. Otherwise, I can't bring in `manager::service::spec::DesiredState` due to:

```
error[E0603]: module `spec` is private
  --> components/sup/src/manager/mod.rs:82:5
   |
82 | use manager::service::spec::DesiredState as SpecDesiredState;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Ok, thanks for the explanation. I think for now this makes sense.